### PR TITLE
Collection support enhancement

### PR DESCRIPTION
**Pull request by MerlinofMines**
Finally, Collection Support is out.

This pull request adds collection support to EasyCommands, enabling the use of Lists of variables.

Functionality includes:
*assign variable to list
*assign list index to value
*retrieval of list values by index
*aggregation of list values, including count, sum, avg, min, max, etc.
*aggregate condition of lists, including all values >, =, < a given value (effectively contains)
*append
*insert
*insert range
*sublist (effectively remove and remove range)
*multi-dimensional list support

### CODE REVIEW

File: EasyCommands/Common/Variables.cs [hunk 1]
```diff
@@ -195,5 +203,81 @@ public class InMemoryVariable : Variable {
...
+            public Variable expectedList;
+            public Variable index;
+
+            public ListIndexVariable(Variable list, Variable i) {
+                expectedList = list;
+                index = i;
+            }
+
+            //TODO: Add Lookup by string support (Dictionary)
+            //TODO: Add List support!  If index is list then return list containing at all requested indexes.  if empty, return expectedList
```
Comment (MerlinofMines) [inline EasyCommands/Common/Variables.cs:229]: This comment is actually addressed

File: EasyCommands/CommandParsers/ParameterParsers.cs [hunk 2]
```diff
@@ -20,11 +20,13 @@
 namespace IngameScript {
     partial class Program {
         //Internal (Don't touch!)
-        private Dictionary<String, List<CommandParameter>> propertyWords = new Dictionary<string, List<CommandParameter>>();
+        Dictionary<String, List<CommandParameter>> propertyWords = new Dictionary<string, List<CommandParameter>>();
+
+        string[] separateTokens = new[] { "(", ")", "[", "]", ",", "+", "*", "/", "!", "^", ".." };
```
Comment (pandincus) [inline EasyCommands/CommandParsers/ParameterParsers.cs:25]: we don't need one for modulo?

**Review by pandincus — APPROVED** [review]
this is amazing

File: EasyCommands/CommandParsers/ParameterParsers.cs (hunk 2 above)
Comment (MerlinofMines) [reply EasyCommands/CommandParsers/ParameterParsers.cs:25]: good call out! Ya we can add modulo :D

**Pull request merged**